### ISSUE TITLE
merge_sort

### ISSUE THREAD

**Issue by ntrost57** [ROCmSoftwarePlatform/rocPRIM#11]
Does device wide merge_sort support in-place sorting (keys_input == keys_output and values_input == values_output)?
What about radix sort? I guess it does not support in-place sorting, but I could not find anything in the docs about it.

**Comment by jszuppe** [comment]
Unless, it's explicitly stated otherwise, input and output iterators must point to separate memory (separate ranges).

**Comment by ntrost57** [comment]
Thank you for your answer. Is radix_sort and/or merge_sort stable? I cannot find anything in the docs about it, however, in the tests, you are comparing to std::stable_sort.

**Comment by jszuppe** [comment]
Radix sort is stable. Merge sort is not.

**Comment by ntrost57** [comment]
Thank you very much! I have one more question:
Using zip_iterator to sort multiple values with the same key is straight forward. What about using zip_iterator with double buffers?

**Comment by jszuppe** [comment]
> What about using zip_iterator with double buffers?

I don't think that's possible. `double_buffer` class template takes value type (`T`) not iterator type. I guess you may try modifying it to operate on iterators instead. You can always try to specialize it for `zip_iterator`.

**Comment by ntrost57** [comment]
I am not sure if that is possible, with integers `zip_iterator` has `value_type` `tuple<int*,int*>` but `radix_sort_impl()` requires value_type `tuple<int, int>*` for buffers. The integer arrays packed with `zip_iterator` are not necessarily continuous in memory.

**Comment by jszuppe** [comment]
Yes, right now it's not possible. It would require some changes to radix sort implementation (it's doable). Right now there are not plans to do them.
